pkg/services/notifications/notifications.go

```diff
@@ -39,7 +43,23 @@ var tmplSignUpStarted = "signup_started"
 var tmplWelcomeOnSignUp = "welcome_on_signup"
 
 func ProvideService(bus bus.Bus, cfg *setting.Cfg, mailer Mailer, store TempUserStore) (*NotificationService, error) {
+	netTransport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			Renegotiation: tls.RenegotiateFreelyAsClient,
+		},
+		Proxy: http.ProxyFromEnvironment,
+		Dial: (&net.Dialer{
+			Timeout: 30 * time.Second,
```

Comment:
What is the meaning of the hardcoded 30s dial timeout (and 5s TLS handshake timeout) in relation to the `http.Client` timeout?

Are they additive, meaning in some cases requests could take 35s longer than the configured timeout? Or does the client timeout encompass the whole request?

If the `http.Client` timeout overrides all, which I **suspect** it does - could we consider relying on that purely and dropping the other timeouts?

Reply:
```
// Timeout specifies a time limit for requests made by this
// Client. The timeout includes connection time, any
// redirects, and reading the response body. The timer remains
// running after Get, Head, Post, or Do return and will
// interrupt reading of the Response.Body.
```

> What is the meaning of the hardcoded 30s dial timeout (and 5s TLS handshake timeout) in relation to the http.Client timeout?

The dial is the timeout for establishing the TCP connection whereas the timeout on the HTTP client is for the HTTP connection. 

I cite the documentation on the `Timeout` field as it illustrates that HTTP client connection does encompass everthing. Indeed the `http.Client` timeout means a maximum of 30s for _everything_ but I wouldn't drop the others as they mean something in their own merit e.g. a TCP connection establishing taking 25s, then 3s for the HTTP call would succeed.

To follow your suggestion, I would need to set all Timeouts to 0s effectively removing the timeouts for those other operations. I briefly paused and thought about that as I putting this PR together but chose not to for two reasons:

1. Alerting is probably the biggest user of the notifications service yes, but I couldn't quickly tell if it's the _only_ user. There's a lot of alerting integrations this code depends to and I'm not sure what kind of side-effects could that code change bring.

2. The problem that I'm trying to solve is very simplistic, more time for HTTP webhooks to succeed - making the suggested change yields no direct benefit (IMO) to that.

If you feel strongly about this - I'm happy to open a separate PR with the change but I would rather not do it under the assumption that this hasn't been a problem.

Reply:
Nah, don't worry about it, just felt the duplicated 30 seemed unnecessary. It's probably not worth removing now.

Reply:
I did consider asking the same question Alex, but I saw it was just copied code. I am interested in understanding if there was a reason for reducing the TLS handshake timeout from a default of 10 seconds to 5 seconds. The TCP connection handshake (dialer) timeout is the same as the default, which is 30 seconds.